### DISCUSSION TITLE
Enter-PSSession and Random Shenanigans

### DISCUSSION THREAD

**Discussion by feamsr00** [JanDeDobbeleer/oh-my-posh#3416] [General]
Firstly, an acceptable answer might very well be "Thats crazy, Your crazy. Just stop it" (I won't, but it's an acceptable answer :)

I've setup OMP on a remote machine and then connected to it via `Enter-PSSession`

![image](https://user-images.githubusercontent.com/983484/215238451-337775be-6166-4152-a85c-954e40de73e3.png)

This adds `[<ComputerName>]: ` to the start of the prompt line. I'd like to know if there's a way OMP can detect its running via remoting. If it can, I think I can configure this theme to hide some segment or at least alter glyphs to be more "integrated" with the remoting prompt.

Is this crazy or *crazy*?

I'm using the `Tokyo` config.

**Comment by JanDeDobbeleer** [comment]
@feamsr00 can you confirm that in this case the `$PSSenderInfo` variable is not `$null`?

**Reply by JanDeDobbeleer** [reply]
@feamsr00 according to the web, you can so something like this:

```powershell
function Set-EnvVar {
    if ( (Test-Path variable:PSSenderInfo)
        -and ($Null -ne $PSSenderInfo)
        -and ($PSSenderInfo.GetType().Name -eq 'PSSenderInfo') ) {
        $env:PSSESSION=1
    } else {
        Remove-Item -Path Env:\PSSESSION -Force -ErrorAction SilentlyContinue
    }
}

New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVar' -Scope Global -Force
```

Adding this after initialising oh-my-posh will make `{{ if .Env.PSSESSION }}` available in templates and you can switch on it.

**Reply by feamsr00** [reply]
@JanDeDobbeleer Briliant!

With PsSession
![image](https://user-images.githubusercontent.com/983484/215681929-667eeff0-f076-46ac-b9c3-de0977f3cc32.png)

Without
![image](https://user-images.githubusercontent.com/983484/215684694-b90bcb10-5c29-450b-b89e-f0022ab4be39.png)

Changed the order of *user from host* to *host with user* to make it a little easier to write the if conditional.

Added this code to my theme\config
``` json
      "segments": [
        {
          "foreground": "#7eb8da",
          "style": "plain",
          "template": "{{ if .Env.PSSESSION }}:[{{else}}\u250f[<#ffffff>\ufcbe</> {{ .HostName }} {{end}}with <#ffffff>\uf508</> {{ .UserName }}]",
          "type": "session"
        },
```
(with PSSession looks a little odd ` :[`, but I felt like mirroring is the best aesthetic you're going to get in this case)

Thanks again!
